source-postgres/discovery.go

```diff
@@ -49,15 +53,15 @@ func (db *postgresDatabase) DiscoverTables(ctx context.Context) (map[string]*sql
 	// using a map from fully-qualified "<schema>.<name>" table names to
 	// the corresponding DiscoveryInfo.
 	var tableMap = make(map[string]*sqlcapture.DiscoveryInfo)
+	for _, table := range tables {
+		var streamID = sqlcapture.JoinStreamID(table.Schema, table.Name)
+		tableMap[streamID] = table
+	}
 	for _, column := range columns {
 		var streamID = sqlcapture.JoinStreamID(column.TableSchema, column.TableName)
 		var info, ok = tableMap[streamID]
 		if !ok {
-			info = &sqlcapture.DiscoveryInfo{
-				Schema:    column.TableSchema,
-				Name:      column.TableName,
-				BaseTable: true, // PostgreSQL discovery queries only ever list 'BASE TABLE' entities
-			}
+			continue
```

Comment:
I think this should only happen if its a column in a system table, if I'm reading things right. If not, it might be worth logging something here since (I assume) it should never happen otherwise.

Reply:
A system table or a child partition, now. This is basically the same logic we already used for primary-key aggregation: if the table isn't already in the list then an earlier discovery step has decided we don't care about it and should ignore this information.

Logging whenever this happens would be unusably spammy even for debug level IMO.